handlers/register.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/speedrun-website/leaderboard-backend/database"
+	"github.com/speedrun-website/leaderboard-backend/graph/model"
+	"github.com/speedrun-website/leaderboard-backend/utils"
+)
+
+func RegisterHandler(c *gin.Context) {
+	var registerValue model.Register
+
+	if err := c.Bind(&registerValue); err != nil {
+		log.Println("Unable to bind value", err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	db, err := database.GetDatabase()
+
+	if err != nil {
+		log.Println("Unable to connect to database", err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	var alreadyExist model.User
+	result := db.Where(model.User{Email: registerValue.Email}).Find(&alreadyExist)
+
+	if result.RowsAffected != 0 {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "An account with this email already exists",
+		})
+		return
+	}
+
+	hash, _ := utils.HashAndSalt([]byte(registerValue.Password))
+
+	db.Create(model.User{
+		Username: registerValue.Username,
+		Email:    registerValue.Email,
+		Password: hash,
+	})
+
+	c.JSON(http.StatusOK, gin.H{
+		"message": "ok",
```

Comment:
We'll have to decide how to handle it as a team, but in my experience creating a resource should result in returning the created resource.

Reply:
When you create a resource, the standard is to return a 201 created response including the `location` header that points to where the resource can be accessed. If you want to also return the created resource that's fine too but not strictly necessary.

Reply:
I'm fine with the 201 and `location` header pattern too.

Reply:
201 + location sounds good. I will return the created ressource too because it's convinient for the frontend.